### PR TITLE
Added permission for Bluetooth Scan and Connect, and update compile and target sdk to 31

### PR DESCRIPTION
**Pull request by ateebakhtar**

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -7,11 +7,11 @@ apply plugin: 'kotlin-android-extensions'
 apply plugin: 'kotlin-kapt'
 
 android {
-    compileSdkVersion 29
+    compileSdkVersion 31
     defaultConfig {
         applicationId "com.mazenrashed.universalbluetoothprinter"
-        minSdkVersion 19
-        targetSdkVersion 29
+        minSdkVersion 23
```
Comment (hamzaahmedkhan) [inline app/build.gradle]: minimum Rider app ka 23 hai?

Comment (hamzaahmedkhan) [reply app/build.gradle]: don't make it 23, because it is an open-source project and you are restricting it for wider audience

Comment (hamzaahmedkhan) [reply app/build.gradle]: 21 is fine

**Review by arbazpirwani — APPROVED** [review]

File: app/build.gradle (hunk 1 above)
Comment (ateebakhtar) [reply app/build.gradle]: Changed to 21

**Pull request merged**